### PR TITLE
samples: Bluetooth: encrypt llpm sample

### PR DESCRIPTION
**Pull request by martintv**
It makes most sense to demonstrate llpm on an ecrypted link. Also setting down conection interval to the smallest bluetooth allows. 7.5ms. Since encyrpting the link takes quite a lot of packets back and forth, this will speed up setup. But also it makes sense that the sample compares the best bluetooth can do with the best llpm can do.

### CODE REVIEW

File: samples/bluetooth/llpm/src/main.c [hunk 1]
```diff
@@ -414,10 +412,24 @@ static void test_run(void)
 	}
 }
 
+void security_changed(struct bt_conn *conn, bt_security_t level,
+				 enum bt_security_err err)
+{
+	printk("Security changed: level %i, err: %i\n", level, err);
+
+	/*Start service discovery when link is encrypted*/
```
Comment (jori-nordic) [inline samples/bluetooth/llpm/src/main.c:420]: can you add a check that `err` is non-zero and disconnect/assert if that's the case?

Comment (martintv) [reply samples/bluetooth/llpm/src/main.c:425]: done

**Comment by martintv** [comment]
image
capture from running sample in bsim on top of this PR

**Comment by jori-nordic** [comment]
@martintv you might want to pass the cmdline option `-RealEncryption=1` to the bsim devices, else bt_encrypt will not really encrypt the data, that's why it appears in cleartext after the `LL_START_ENC`s are exchanged.

**Review by jori-nordic — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Pull request merged**